### PR TITLE
Retry downloading the toolchain

### PR DESCRIPTION
**Pull request by imikushin**
Address #356.

### CODE REVIEW

File: jolt-core/src/host/toolchain.rs [hunk 1]
```diff
@@ -22,13 +25,40 @@ pub fn install_toolchain() -> Result<()> {
     let toolchain_url = toolchain_url();
 
     let rt = Runtime::new().unwrap();
-    rt.block_on(download_toolchain(&client, &toolchain_url))?;
+    println!("Downloading toolchain (retrying up to {DOWNLOAD_RETRIES} times)");
```
Comment (sragss) [inline jolt-core/src/host/toolchain.rs:28]: nit: rm print.

File: jolt-core/src/host/toolchain.rs [hunk 2]
```diff
@@ -22,13 +25,40 @@ pub fn install_toolchain() -> Result<()> {
...
+        F: Fn() -> E,
+        E: Future<Output=Result<T>>,
+{
+    for i in 0..times {
+        println!("Attempt {}/{}", i + 1, times);
+        match f().await {
+            Ok(t) => return Ok(t),
+            Err(e) => {
+                let timeout = delay_timeout(i, base_ms);
+                println!("Error: {}. Retrying in {}ms", e, timeout);
```
Comment (sragss) [inline jolt-core/src/host/toolchain.rs:49]: nit: `println!("Toolchain download error {i}/{times}: {e}. Retrying in {timeout}ms")`

**Review by sragss — COMMENTED** [review]
Looks great. Some nits to keep the prints clean.

Could you also run `cargo fmt`?

File: jolt-core/src/host/toolchain.rs (hunk 1 above)
Comment (imikushin) [reply jolt-core/src/host/toolchain.rs]: Done

File: jolt-core/src/host/toolchain.rs (hunk 2 above)
Comment (imikushin) [reply jolt-core/src/host/toolchain.rs]: Done

**Comment by sragss** [comment]
LGTM

**Pull request merged**